### PR TITLE
Fix logical acid-base homeostasis related axioms

### LINKED ISSUE #7534: Acidosis & Alkalosis logical definition causes lots of bad inference
**HPO term**

acidosis: 'has part' some 
    ('increased amount'
     and ('characteristic of' some 
        ('chemical entity'
         and ('part of' some 'bodily fluid')
         and ('has role' some acid)))
     and ('has modifier' some abnormal))

Leads to classification of increased levels of substances in bodily fluids that recorded as acids by CHEBI e.g.- Hyperglutaminemia
is an inferred subclass due to it being glutamine being an amino acids (it's actually also inferred to be Akalosis as well due.
'has part' some 
    ('increased amount'
     and ('characteristic of' some 
        (glutamine
         and ('part of' some blood)))
     and ('has modifier' some abnormal))

**Suggested revision of logical definition**

The definition should not be based on the nature of chemicals but on abnormal pH of the bodily fluid => 

'has part' some ('increased acidity' and ('characteristic of' some 'body fluid') and (has_modifier some 'abnormal')

ditto Alkalosis.

### PR DESCRIPTION
**Pull request by rays22**
This commit intends to
- remove some axioms that result in incorrect inferences,
- import acidity-related PATO qualities,
- fix incorrect EQs for 'Acidemia' and 'Alkalemia'. If applied, this commit will fix #7534.

### CODE REVIEW

File: src/ontology/hp-edit.owl [hunk 1]
```diff
@@ -168662,7 +168660,7 @@ AnnotationAssertion(dc:date <http://purl.obolibrary.org/obo/HP_0032368> "2019-02
...
 AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/HP_0032368> "Acidemia")
-EquivalentClasses(<http://purl.obolibrary.org/obo/HP_0032368> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/BFO_0000051> ObjectIntersectionOf(<http://purl.obolibrary.org/obo/PATO_0000470> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000052> ObjectIntersectionOf(<http://purl.obolibrary.org/obo/CHEBI_24431> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/BFO_0000050> <http://purl.obolibrary.org/obo/UBERON_0001977>) ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000087> <http://purl.obolibrary.org/obo/CHEBI_37527>))) ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0002573> <http://purl.obolibrary.org/obo/PATO_0000460>))))
+EquivalentClasses(<http://purl.obolibrary.org/obo/HP_0032368> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/BFO_0000051> ObjectIntersectionOf(<http://purl.obolibrary.org/obo/PATO_0001844> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000052> <http://purl.obolibrary.org/obo/UBERON_0000178>) ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0002573> <http://purl.obolibrary.org/obo/PATO_0000460>))))
```
Comment (matentzn) [inline src/ontology/hp-edit.owl:168663]: Should acidemia be defined in terms of "decreased acidity"? (I usually do not look at content, but since I want this to be merged asap :P)

**Comment by pnrobinson** [comment]
@rays22 @matentzn 
Acidosis now has two defs
```
'has part' some 
    (acidity
     and ('characteristic of' some 'bodily fluid')
     and ('has modifier' some abnormal))
```
and
```
'has part' some 
    (quality
     and ('characteristic of part of' some 'metabolic process')
     and ('has modifier' some abnormal))
```
The first stanza should probably be this
```
'has part' some 
    (acidity
     and ('characteristic of' some 'bodily fluid')
     and ('has modifier' some INCREASED))
```
because too little acidity is also abnormal (that is alkalemia)
I do not think the second part is really adding very much, and we usually do not have multiple stanzas in these definitions -- why not delete or use an "and"

Alkalemia has the exact same definition and seems to be a copy paste error.
It could be
```
'has part' some 
    (acidity
     and ('characteristic of' some 'bodily fluid')
     and ('has modifier' some DECREASED))
```
OR
```
'has part' some 
    (alkalinity
     and ('characteristic of' some 'bodily fluid')
     and ('has modifier' some INCREASED))
```

File: src/ontology/hp-edit.owl (hunk 1 above)
Comment (rays22) [reply src/ontology/hp-edit.owl:168663]: It is kind of confusing, because increased acidity means [lower pH](https://en.wikipedia.org/wiki/PH), but 'Acidemia' is [An abnormally low blood pH (usually defined as less than 7.35).](https://www.ebi.ac.uk/ols4/ontologies/hp/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0032368)

'pH' is indeed a synonym of [PATO:0001842 acidity](https://www.ebi.ac.uk/ols4/ontologies/pato/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FPATO_0001842), but the textual definition of [PATO:0001844 increased acidity](https://www.ebi.ac.uk/ols4/ontologies/pato/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FPATO_0001844?lang=en)) refers to higher acidity. People who measure acidity and pH in laboratories and report related phenotypes are aware of the negative correlation between acidity and pH, and when they say 'increased acidity' it should be unambiguous what it means in terms of pH.
In summary, acidemia is correctly defined as increased acidity here.

**Comment by matentzn** [comment]
@rays22 can you provide the human readable versions of the new defs here for Peter to review?

**Comment by rays22** [comment]
> @rays22 @matentzn Acidosis now has two defs
> 
> ```
> 'has part' some 
>     (acidity
>      and ('characteristic of' some 'bodily fluid')
>      and ('has modifier' some abnormal))
> ```
> 
> and
> 
> ```
> 'has part' some 
>     (quality
>      and ('characteristic of part of' some 'metabolic process')
>      and ('has modifier' some abnormal))
> ```

The two defs above come from the assertions of 'HP:0001941 Acidosis' as being a sub-lcass of HP:0001939 Abnormality of metabolism/homeostasis and HP:0004360 Abnormality of acid-base homeostasis. My PR does not change these subclass assertions. In other words, these 

> The first stanza should probably be this
> 
> ```
> 'has part' some 
>     (acidity
>      and ('characteristic of' some 'bodily fluid')
>      and ('has modifier' some INCREASED))
> ```

Your suggestion does not look correct for several reasons. First of all, this stanza is inferred (logically correctly) by the reasoner from the taxonomy  and not editable at the level of this term. Secondly, the text definition of HP:0001941 Acidosis goes to great lengths to explain why Acidosis is more complex than just final outcome of an observed pH. Your suggested axiom does not express that complexity. 

> because too little acidity is also abnormal (that is alkalemia) I do not think the second part is really adding very much, and we usually do not have multiple stanzas in these definitions -- why not delete or use an "and"
> 
> Alkalemia has the exact same definition and seems to be a copy paste error. It could be

No, it is not a copy paste error. It is correctly inferred by the reasoner for Acidemia being a subclass of HP:0004360 Abnormality of acid-base homeostasis. Please note, that 'increased acidity' is a type of acidity both ontologically and by common usage in laboratories. This is correctly modelled in PATO as increased acidity is a subclass of acidity.

> 
> ```
> 'has part' some 
>     (acidity
>      and ('characteristic of' some 'bodily fluid')
>      and ('has modifier' some DECREASED))
> ```
> 
> OR
> 
> ```
> 'has part' some 
>     (alkalinity
>      and ('characteristic of' some 'bodily fluid')
>      and ('has modifier' some INCREASED))
> ```

The text definition of HP:0001941 Acidosis goes to great lengths to explain that it is related to the biological processes that result in abnormal acidity or alkalinity. As GO uses PATO 'acidity (PATO:0001842)' to define the related biological processes (see GO:0006885 regulation of pH), it makes sense to use acidity and its child terms here for *interoperability with a vast amount of experimental data that have been annotated by GO terms.*

I would suggest that the axioms of superclasses of Acidosis to be the subject of another ticket and so that this PR can fix the incorrectly inferred classification of 347 terms as being types of acidosis or alkalosis, or both, e.g. HP:0034604 Elevated circulating DOPA concentration.

**Comment by rays22** [comment]
This is an illustration of the kind of problem that this PR intends to fix.
* HP:0034604 Elevated circulating DOPA concentration is incorrectly inferred as being both acidosis and alkalosis.

Elevated_circulating_DOPA_concentration-Superclasses

**Review by matentzn — APPROVED** [review]
Thanks for the explanations @rays22, makes sense to me!

**Comment by pnrobinson** [comment]
I do not understand at all, but the definitions just seem wrong. Could we discuss?

**Comment by matentzn** [comment]
You dont think that this definition makes sense for Alkalemia?

image

**Comment by rays22** [comment]
> @rays22 can you provide the human readable versions of the new defs here for Peter to review?

1. The corrected EQ for the [Abnormality of acid-base homeostasis](http://purl.obolibrary.org/obo/HP_0004360)
```
'has part' some 
    (acidity
     and ('characteristic of' some 'bodily fluid')
     and ('has modifier' some abnormal))
```

means that HP:0004360 is a phenotypic attribute that encompasses the manifestation of any abnormal acidity (pH) of a bodily fluid. This logical definition includes manifestations of *both increased and decreased acidity* (and alkality) of bodily fluids irrespective of what causes the deviation from the homeostatic state of the acid-base balance.
The EQ must be true for any class that is declared a subclass of HP:0004360. These 57 subclasses have been asserted outside of this PR. Check if those subclass assertions are correct: 
Abnormality_of_acid-base_homeostasis-corrected

You can add distinguishing characteristics to its subclasses if needed.
2. The corrected EQ for [HP:0032368 Acidemia](https://www.ebi.ac.uk/ols4/ontologies/hp/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0032368):
```
'has part' some 
    ('increased acidity'
     and ('characteristic of' some blood)
     and ('has modifier' some abnormal))
```

means that it is an abnormally decreased blood pH.
3. The corrected EQ for [HP:0032369 Alkalemia](https://www.ebi.ac.uk/ols4/ontologies/hp/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0032369)
```
'has part' some 
    ('decreased acidity'
     and ('characteristic of' some blood)
     and ('has modifier' some abnormal))
```
means that it is an abnormally increased blood pH.

**Comment by pnrobinson** [comment]
Sorry if I am being dense, but GitHub is not a great place to discuss changes to OWL files.
Do you mean that we now have 

'has part' some 
    ('increased acidity'
     and ('characteristic of' some blood)
     and ('has modifier' some abnormal))

as the characteristic definition for the acidity terms and no longer "abnormal acidity" (the latter is incorrect)?

**Comment by rays22** [comment]
> Do you mean that we now have
> 
> 'has part' some ('increased acidity' and ('characteristic of' some blood) and ('has modifier' some abnormal))
> 
> as the characteristic definition for the acidity terms 

No, this is the definition of only 'Acidemia' (very well matching the text definition of [An abnormally low blood pH (usually defined as less than 7.35).](https://www.ebi.ac.uk/ols4/ontologies/hp/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0032368) ) and its subclasses:
Acidemia-corrected


>and no longer "abnormal acidity" (the latter is incorrect)?

I am not sure if I understand your question. 'increased acidity' is a more specific type of 'acidity' and the definition explicitly states 'abnormal' as a qualifier. In summary, 'Acidemia' is still abnormal acidity of blood. It is blood and not bodily fluid, because that is what the text definition states. However, blood is a bodily fluid and it is represented as such in Uberon.

Another way to think about this EQ and its superclasses is that it means that every acidemia is an abnormal balance of acids and bases in a bodily fluid, but not every abnormal balance of acids and bases is an acidemia.

**Comment by pnrobinson** [comment]
@rays22 
perhaps it is clearer to state that both pH = 6.9 and 8.2 are abnormal acidity (amount of acid), but only 6.9 is increased acidity (and acidemia). 8.2 is insufficient acidity

**Comment by rays22** [comment]
> @rays22 perhaps it is clearer to state that both pH = 6.9 and 8.2 are abnormal acidity (amount of acid), but only 6.9 is increased acidity (and acidemia). 8.2 is insufficient acidity

@pnrobinson ,
The current text definition may not be clear for everybody, but my aim in this PR is limited to fixing the EQs that engender incorrect inferences. I will let you and the experts define the clinical parameters for normal and abnormal blood/bodily fluid pH ranges. The current text definition of HP:0032368 references [PMID:24381489](https://pubmed.ncbi.nlm.nih.gov/24381489/), and Figure 1 in that paper provides some detailed guidelines and clarifications for pH ranges for acidosis and alkalosis.

**Comment by pnrobinson** [comment]
@rays22 I think we are misunderstanding each other, but as I say I do not think these definitions make sense. Can we zoom?

**Comment by rays22** [comment]
> @rays22 I think we are misunderstanding each other, but as I say I do not think these definitions make sense. Can we zoom?

@pnrobinson ,
OK, I can set up a zoom meeting and I have sent you a message about it on the Monarch slack channel.
